llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -108,6 +109,21 @@ class AArch64AsmPrinter : public AsmPrinter {
 
   StringRef getPassName() const override { return "AArch64 Assembly Printer"; }
 
+  bool doInitialization(Module &M) override {
+    if (AsmPrinter::doInitialization(M))
+      return true;
+
+    if (TM.getTargetTriple().isOSBinFormatELF()) {
```

Comment:
Nit: AArch64MachineFunctionInfo.cpp contains `hasELFSignedGOTHelper`, and it might be worth moving that helper somewhere so it can be accessed both from there and from here.

Feel free to ignore

Reply:
I did look at that. Most of the suitable header files look like they're meant specifically for one class and it wouldn't be right to add random utility functions there. Turning it into a member function of some class also didn't work out, since it can't be in `AArch64FunctionInfo` since we need to be able to access it when we're not processing any function, and it shouldn't be in `AArch64Subtarget` since it's useful to be able to access it before we've created a `AArch64Subtarget`. But looking again for somewhere to put it as a freestanding function one that is justifiable is `AArch64PointerAuth.h` so I've done that now, thanks.